docs(web): add new page for tabdatalist (#340)
appbaseio/Docs: content/docs/reactivesearch/v3/list/tabdatalist.md

---
title: 'TabDataList'
meta_title: 'TabDataList'

> meta_description: 'TabDataList creates a multiple selection based list UI component that is connected to a database field.'

Reviewer: ```suggestion
meta_description: 'TabDataList creates a multiple selection based list UI component that is connected to a search index field.'
```

        {
            ({
                data,
                value,
                handleChange,
            }) => (
                // return UI to be rendered
            )
        }
</TabDataList>
```
## Demo

<br />

<iframe src="https://codesandbox.io/embed/github/appbaseio/reactivesearch/tree/next/packages/web/examples/TabDataList" style="width:100%; height:500px; border:0; border-radius: 4px; overflow:hidden;" sandbox="allow-modals allow-forms allow-popups allow-scripts allow-same-origin"></iframe>

## Extending

`TabDataList` component can be extended to

1. customize the look and feel with `className`, `style`,
2. update the underlying DB query with `customQuery`, `defaultQuery`,

```js
<TabDataList
  ...
  className="custom-class"
  style={{"paddingBottom": "10px"}}
  defaultQuery={
    function(value, props) {
      return {
        timeout: "1s"
      }
    }
  }
  customQuery={
    function(value, props) {
      return {
        query: {
            match: {
                data_field: "this is a test"
            }
        }
      }
    }
  }
/>
```

-   **className** `String`
    CSS class to be injected on the component container.
-   **style** `Object`

>     CSS styles to be applied to the **SingleDataList** component.

Reviewer: ```suggestion
    CSS styles to be applied to the **TabDataList** component.
```

-   **customQuery** `Function`
    takes **value** and **props** as parameters and **returns** the data query to be applied to the component, as defined in Elasticsearch Query DSL.

>     `Note:` customQuery is called on value changes in the **SingleDataList** component as long as the component is a part of `react` dependency of at least one other component.

Reviewer: ```suggestion
    `Note:` customQuery is called on value changes in the **TabDataList** component as long as the component is a part of `react` dependency of at least one other component.
```
